docker-build-and-publish/action.yml

```diff
@@ -0,0 +1,32 @@
+name: Build and Publish TimescaleDB Docker Image
+description: Build a TimescaleDB Docker image ready for publishing
+inputs:
```

Comment:
should we have an optional input for push? You may want to build an image without pushing (for testing the build process)

Reply:
Added one, and one for the tools version. Not particularly fond of the `Dockerfile`, but we can refactor and optimize later.